packages/ui/src/models/camel/camel-route-resource.ts

```diff
@@ -132,7 +133,8 @@ export class CamelRouteResource implements CamelResource, BeansAwareResource {
      * as this mean that they can be used only as a consumer.
      */
     return (item: ITile) => {
-      return !item.tags.includes('consumerOnly');
+      return (item.type !== CatalogKind.Kamelet && !item.tags.includes('consumerOnly')) || 
+      (item.type === CatalogKind.Kamelet && !item.tags.includes('source'));
```

Comment:
The conditions look good :ok_hand: 

Do you think it would be possible to add a test for this method in the `camel-route.test.ts` file?

Certainly, we would need to have a mock for the catalog but it's not available yet, so we could create 3 scenarios and assert that we get a function on each of them, after that, once we have the catalog mock, we can do better assertions against the real tiles.

Reply:
The structure could be:

```
  describe('getCompatibleComponents')
    it(`scenario for from`)
    it(`scenario for InsertSpecialChil`)
    it(`remaining scenario`)
```

note the messages are simply placeholder, we would need a description that explains the test case appropriately

Reply:
I get it. I'll try to add those tests in some time!

Reply:
Do you mean something like this ?
![image](https://github.com/KaotoIO/kaoto-next/assets/73224329/5153978e-6ecf-4cd0-a595-ca07aa9bacde)

Reply:
Yes, that's a good start, but remember to provide a descriptive name for the `it`.

* `it('should not provide `isProducerOnly` components')`
* `it('should provide components')`
* `it('should provide processors')`
* `it('should provide `source` kamelets')`

Reply:
Then the assertion would be to receive a function out of the `getCompatibleComponents` method. Later on, when we provide a catalog mock, we'll change those assertions to something more specific, by testing which components we get in response.